perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/AbstractApplicationExpirationRulesModule.java

```diff
@@ -0,0 +1,54 @@
+package cz.metacentrum.perun.core.implApi.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.PerunBean;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Jakub Hejda <Jakub.Hejda@cesnet.cz>
+ */
+public abstract class AbstractApplicationExpirationRulesModule<T extends PerunBean> extends AttributesModuleAbstract implements AttributesModuleImplApi {
+	public static final String applicationWaitingForEmailVerificationKeyName = "emailVerification";
+	public static final String applicationIgnoredByAdminKeyName = "ignoredByAdmin";
+	private static final Pattern daysToExpirationPattern = Pattern.compile("^\\d+$");
+
+	public void checkAttributeSyntax(PerunSessionImpl sess, T entity, Attribute attribute) throws WrongAttributeValueException {
+
+		Map<String, String> attrValue;
+
+		//For no value is correct (it means no rules)
+		if (attribute.getValue() == null) return;
+
+		//save value to map attrValue
+		attrValue = attribute.valueAsMap();
+
+		//Same for empty HashList
+		if (attrValue.isEmpty()) return;
```

Comment:
I believe ``attribute.valueAsMap()`` wouldn't throw a null pointer, since you can cast null to any type. You could simplify it like this:
```java
Map<String, String> attrValue = attribute.valueAsMap();

if (attrValue == null || attrValue.isEmpty()) return;
```

Reply:
Simplified